ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -2427,7 +2430,7 @@ protected RelNode executeProgram(RelNode basePlan, HepProgram program,
       // Create planner and copy context
       HepPlanner planner = new HepPlanner(program,
           basePlan.getCluster().getPlanner().getContext());
-
+      planner.addListener(new RuleEventLogger());
```

Comment:
Should we add a check the log level setting before adding this listener like
```
if (LOG.isDebugEnabled()) {
  planner.addListener(new RuleEventLogger());
}
```
It may seems to be a preliminary optimization, what do you think?

Reply:
`RuleEventLogger` is using a different logger (not LOG) so cannot easily put this check here. Moreover, the performance benefit is unclear so it might be better to leave it is as is. WDYT?

Reply:
If that's the case you can leave it is as is.